rsyslog/undefined_field/undefined_field.go

```diff
@@ -389,6 +387,51 @@ func processUndefinedAndEmpty(input map[string]interface{}, hasDefinedFields, ha
 	return inputWasModified
 }
 
+// The given rawStr is only used for logging purposes.  The given
+// input map is modified in-place.  The return value is false if there
+// were errors, or no changes were made to the given input.  If the
+// return value is true, the caller needs to handle outputting the
+// new fields.
+func processFields(rawStr string, input map[string]interface{}) bool {
+	changes := false
+	undefString, undefMap, err := processUndefinedAndMaxNumFields(input)
+	if err != nil {
+		// error marshalling undefined fields to JSON
+		if cfg.Debug {
+			fmt.Fprintf(logfile, "Unable to convert undefined fields to JSON string: %v : rawStr: %s\n", err, rawStr)
+		}
+		fmt.Println(noChanges)
+		return changes
+	}
+	if len(undefString) > 0 || undefMap != nil {
+		changes = true
+	}
+	if undefMap != nil {
+		if processUndefinedAndEmpty(undefMap, false, true) {
+			changes = true
```

Comment:
No harm at all, but changes is already set to true.  It confused me for a moment if there's a case to reset changes to false by any chance... :)  I guess no need to check the return value from processUndefinedAndEmpty...?  (same for line 416)

Reply:
ok - yes, I changed 410 to ignore the return value since it is redundant

but 416 (now line 414) - I think I still need to set `changes = true` at line 415 - I think `changes` can be false at line 414

Reply:
Ah, you are right!  Sorry for the noise...

/lgtm